atom/browser/api/atom_api_power_monitor.cc

```diff
@@ -39,6 +60,22 @@ void PowerMonitor::OnResume() {
   Emit("resume");
 }
 
+void PowerMonitor::QuerySystemIdleState(v8::Isolate* isolate,
+                                        int idle_threshold,
+                                        const ui::IdleCallback& callback) {
+  if (idle_threshold > 0) {
+    ui::CalculateIdleState(idle_threshold, callback);
+  } else {
+    isolate->ThrowException(v8::Exception::TypeError(
+        mate::StringToV8(isolate,
+          "Invalid idle threshold, must be greater than 0")));
```

Comment:
I know another reviewer suggested this, but are we sure that we want this?

Since we're not publishing the `ui::CheckIdleStateIsLocked()`, if all you want to do is check for system lock it's reasonable to pass a `-1` in for `idle_threshold`.

If we include that method, we can avoid this wart by having a recommended way to test for lock

Reply:
hmm I didn't thought in that way, but also checking `lock` state is somewhat limited per platform support, so passing `-1` doesn't guarantee you can reliable acquire intended state. I think it's rather limit api surface, and expose `CheckIdleStateIsLocked` separately if there's demand for it.